components/Code/index.tsx

```diff
@@ -71,7 +73,7 @@ const Code = (props: CodeProps) => {
   const code = value ?? (Array.isArray(children) ? children[0] : children) ?? '';
 
   const highlightedCode =
-    syntaxHighlighter && typeof syntaxHighlighter === 'function' && code
+    syntaxHighlighter && typeof syntaxHighlighter === 'function' && code && isHydrated
```

Comment:
Applying the syntax highlighting immediately after the first client side render (aka hydration) ensures the SSR and CSR html will match and React will be happy.